papexp/export.py

```diff
@@ -0,0 +1,130 @@
+        categories[item['uid']] = item['name']
+
+
+    c.request('GET', '/api/v1/sync/recipes/', headers=headers)
+    res = c.getresponse()
+    data = res.read()
+
+    recipes = []
+
+    for item in json.loads(data)['result']:
+        c.request('GET', '/api/v1/sync/recipe/'+item['uid']+'/', headers=headers)
+        res = c.getresponse()
+        data = res.read()
+        recipe = json.loads(data)['result']
+        #! https://gist.github.com/mattdsteele/7386ec363badfdeaad05a418b9a1f30a
+        print(recipe['name'])
+        if recipe['photo_large']:
+            recipe['photo'] = recipe['photo_large']
+            addr = recipe['photo_large'][:-4]
+            c.request('GET', "/api/v1/sync/photo/"+addr+"/", headers=headers)
+            res = c.getresponse()
+            data = res.read()
+            photoData = json.loads(data)['result']
+            recipe['photo_url'] = photoData['photo_url']
+
+
+        if recipe['photo'] and recipe['photo_url'] and recipe['photo_url'].startswith('http://uploads.paprikaapp.com.s3.amazonaws.com'):
+        #!if recipe['photo_url']:
+            resp = requests.get(recipe['photo_url'], stream=True)
+            local_file = open('assets/images/recipes/'+recipe['photo'], 'wb')
+            resp.raw.decode_content = True
+            shutil.copyfileobj(resp.raw, local_file)
+            recipe['image_url'] = 'images/recipes/'+recipe['photo']
+
+
+        del recipe['photo_url']
+        del recipe['photo']
+        del recipe['hash']
+        del recipe['photo_hash']
+        del recipe['photo_large']
+
+        recipe['photos'] = []
+
+        if recipe['directions']:
+            directions = recipe['directions'].split('\n')
+            recipe['directions'] = []
+            for direction in directions:
+                if direction != '':
+                    recipe['directions'].append(direction)
+
+        if recipe['ingredients']:
+            ingredients = recipe['ingredients'].split('\n')
+            recipe['ingredients'] = []
+            for ingredient in ingredients:
+                if ingredient != '':
+                    recipe['ingredients'].append(ingredient)
+
+        if recipe['categories']:
+            categoryList = []
+            for category in recipe['categories']:
+                categoryList.append(categories[category])
+            recipe['categories'] = categoryList
+
+        recipes.append(recipe)
+
+    #! this gets all photos
+    c.request('GET', '/api/v1/sync/photos/', headers=headers)
+    res = c.getresponse()
+    data = res.read()
+    photos = json.loads(data)['result']
+    for item in json.loads(data)['result']:
+        c.request('GET', '/api/v1/sync/photo/'+item['uid']+'/', headers=headers)
+        res = c.getresponse()
+        data = res.read()
+        photo = json.loads(data)['result']
+        rec = [x for x in recipes if x['uid'] == photo['recipe_uid']]
+        #! create newphoto dict with uid and filenamd
```

Comment:
is this comment for the line above it?  If so I would move it to be above `rec =`